openpype/plugins/publish/integrate_new.py

```diff
@@ -304,441 +565,245 @@ def register(self, instance):
-                    template_data["frame"] = repre_context["frame"]
-                else:
-                    template_data["udim"] = repre_context["udim"]
-
-                self.log.debug(
-                    "test_dest_files: {}".format(str(test_dest_files)))
-
-                dst_collections, remainder = clique.assemble(test_dest_files)
-                dst_collection = dst_collections[0]
-                dst_head = dst_collection.format("{head}")
-                dst_tail = dst_collection.format("{tail}")
+            # Fall back to instance staging dir if not explicitly
+            # set for representation in the instance
+            self.log.debug("Representation uses instance staging dir: "
+                           "{}".format(instance_stagingdir))
+            stagingdir = instance_stagingdir
+        if not stagingdir:
+            raise ValueError("No staging directory set for representation: "
+                             "{}".format(repre))
 
-                index_frame_start = None
+        self.log.debug("Anatomy template name: {}".format(template_name))
+        anatomy = instance.context.data['anatomy']
+        template = os.path.normpath(anatomy.templates[template_name]["path"])
+
+        is_sequence_representation = isinstance(files, (list, tuple))
+        if is_sequence_representation:
+            # Collection of files (sequence)
+            # Get the sequence as a collection. The files must be of a single
+            # sequence and have no remainder outside of the collections.
+            collections, remainder = clique.assemble(files,
+                                                     minimum_items=1)
+            if not collections:
+                raise ValueError("No collections found in files: "
+                                 "{}".format(files))
+            if remainder:
+                raise ValueError("Files found not detected as part"
+                                 " of a sequence: {}".format(remainder))
+            if len(collections) > 1:
+                raise ValueError("Files in sequence are not part of a"
+                                 " single sequence collection: "
+                                 "{}".format(collections))
+            src_collection = collections[0]
+
+            # If the representation has `frameStart` set it renumbers the
+            # frame indices of the published collection. It will start from
+            # that `frameStart` index instead. Thus if that frame start
+            # differs from the collection we want to shift the destination
+            # frame indices from the source collection.
+            destination_indexes = list(src_collection.indexes)
+            destination_padding = len(get_first_frame_padded(src_collection))
+            if repre.get("frameStart") is not None:
+                index_frame_start = int(repre.get("frameStart"))
 
                 # TODO use frame padding from right template group
```

Comment:
@mkolar do you happen to know what the "right template group" would be? :) This is an old to do comment and I'm wondering if I could easily resolve this as I go.

Reply:
Hmm - I can see the entries there - and I also see the Templates in Anatomy however it's not clear how using a custom template name I'd overwrite the frame padding value instead?

Reply:
Hmm. I think we planned to allow defining custom padding in each template category, but eventually removed it to reduce complexity. Looks like it would be safe to remove this TODO

Reply:
You can use `anatomy.templates[template_name]["frame_padding"]`.

Reply:
Not tested yet, but implemented with https://github.com/pypeclub/OpenPype/pull/2898/commits/3e095bc7554a24ef13282ccfd87e0327eb3b8745